sdks/go/pkg/beam/core/runtime/graphx/serialize.go

```diff
@@ -257,22 +257,22 @@ func decodeFn(u *v1.Fn) (*graph.Fn, error) {
 	if u.Fn != nil {
 		fn, err := decodeUserFn(u.Fn)
 		if err != nil {
-			return nil, fmt.Errorf("bad userfn: %v", err)
+			return nil, fmt.Errorf("failed to decode function proto %v, bad userfn: %v", u, err)
 		}
 		fx, err := funcx.New(reflectx.MakeFunc(fn))
 		if err != nil {
-			return nil, fmt.Errorf("bad userfn: %v", err)
+			return nil, fmt.Errorf("failed to decode function proto %v, bad userfn: %v", u, err)
```

Comment:
("failed to decode DoFn %v, bad userfn: %v", u, err)

Reply:
Done, with a little added detail:

("failed to decode DoFn %v, failed to construct userfn: %v", u, err)